File: 404.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int sumOfLeftLeavesHelper(struct TreeNode* root, int isLeft) {
    if (!root) return 0;
    if (!root->left && !root->right) {
        if (isLeft) return root->val;
        return 0;
    }
    return sumOfLeftLeavesHelper(root->left, 1) + sumOfLeftLeavesHelper(root->right, 0);
}

int sumOfLeftLeaves(struct TreeNode* root) {
    return sumOfLeftLeavesHelper(root, 0);
}

File: 404.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        
        def sumOfLeftLeavesHelper(root, left):
            if not root:
                return 0
            elif root.left or root.right:
                return sumOfLeftLeavesHelper(root.left, True) + sumOfLeftLeavesHelper(root.right, False)
            else:
                if left:
                    return root.val
                else:
                    return 0
        
        return sumOfLeftLeavesHelper(root, False)
